cmd/ctr/checkpoint.go

```diff
@@ -80,9 +80,26 @@ var checkpointCommand = cli.Command{
 		}
 		var additionalDescriptors []*descriptor.Descriptor
 		if context.Bool("binds") {
+			stopped := context.Bool("exit")
+			// if the container is still running make sure the pause the container before checkpointing binds
+			// and other resources
+			if !stopped {
+				if _, err := containers.Pause(ctx, &execution.PauseRequest{
```

Comment:
Do you pause a container to snapshot file systems? In this case, we need to pause a container before checkpointing it. Otherwise there is a race, because a container file system can be changed between container.checkpoint and checkpointBinds().

But currently checkpoint() doesn't work for paused containers, there is a commit for runc to handle this case:
https://github.com/opencontainers/runc/pull/1355/commits/73258813d37effc24075c1d6333b18441da18447